zrml/prediction-markets/src/migrations.rs

```diff
@@ -15,7 +15,246 @@
+        let dispute_duration = T::DisputePeriod::get();
+        let oracle_duration = T::ReportingPeriod::get();
+        let deadlines = Deadlines { grace_period: 0_u32.into(), oracle_duration, dispute_duration };
+        let mut new_markets_data: Vec<(Vec<u8>, MarketOf<T>)> = Vec::new();
+        for (key, legacy_market) in storage_iter::<LegacyMarketOf<T>>(MARKET_COMMONS, MARKETS) {
+            total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+            let new_market = Market {
+                creator: legacy_market.creator,
+                creation: legacy_market.creation,
+                creator_fee: legacy_market.creator_fee,
+                oracle: legacy_market.oracle,
+                metadata: legacy_market.metadata,
+                market_type: legacy_market.market_type,
+                period: legacy_market.period,
+                scoring_rule: legacy_market.scoring_rule,
+                status: legacy_market.status,
+                report: legacy_market.report,
+                resolved_outcome: legacy_market.resolved_outcome,
+                dispute_mechanism: legacy_market.dispute_mechanism,
+                deadlines,
+            };
+            new_markets_data.push((key, new_market));
+        }
+        for (key, new_market) in new_markets_data {
+            put_storage_value::<MarketOf<T>>(MARKET_COMMONS, MARKETS, &key, new_market);
+            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+        }
+        log::info!("Completed updates of markets");
+        utility::put_storage_version_of_market_commons_pallet(MARKET_COMMONS_NEXT_STORAGE_VERSION);
+        total_weight
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<(), &'static str> {
+        Ok(())
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade() -> Result<(), &'static str> {
+        let dispute_duration = T::DisputePeriod::get();
+        let oracle_duration = T::ReportingPeriod::get();
+        let deadlines = Deadlines { grace_period: 0_u32.into(), oracle_duration, dispute_duration };
+        for (market_id, market) in storage_iter::<MarketOf<T>>(MARKET_COMMONS, MARKETS) {
+            assert_eq!(
+                market.deadlines, deadlines,
+                "found unexpected deadlines in market. market_id: {:?}, deadlines: {:?}",
+                market_id, market.deadlines
+            );
+        }
+        Ok(())
+    }
+}
+
+impl<T: Config> OnRuntimeUpgrade for MigrateMarketIdsPerBlockStorage<T> {
+    fn on_runtime_upgrade() -> Weight
+    where
+        T: Config,
+    {
+        let mut total_weight = T::DbWeight::get().reads(2);
+
+        if StorageVersion::get::<Pallet<T>>()
+            != PREDICTION_MARKETS_REQUIRED_STORAGE_VERSION_FOR_MIGRATE_MARKET_IDS_STORAGE
+            || utility::get_on_chain_storage_version_of_market_commons_pallet()
+                != MARKET_COMMONS_NEXT_STORAGE_VERSION
```

Comment:
So you would like to execute the migration when `MARKET_IDS_STORAGE` is not already migrated AND when `market-commons` is already migrated. 🤯
So you would like to execute the other migration of `UpdateMarketsForDeadlines` first?

Reply:
yes UpdateMarketsForDeadlines is executed first.